### PR TITLE
net-proxy/mitmproxy-5.1.1: Version Bump

### PR DESCRIPTION
**Pull request by kwhat**
**dev-python/kaitaistruct: Add python3_7**
Bug: https://bugs.gentoo.org/718796
Closes: https://bugs.gentoo.org/718796
Signed-off-by: Alex Barker alex@1stleg.com

**net-proxy/mitmproxy-5.1.1: Version Bump**
Bug: https://bugs.gentoo.org/708442
Bug: https://bugs.gentoo.org/718458
Closes: https://bugs.gentoo.org/708442
Closes: https://bugs.gentoo.org/718458
Signed-off-by: Alex Barker alex@1stleg.com

### CODE REVIEW

File: net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild [hunk 1]
```diff
@@ -54,6 +57,9 @@ python_prepare_all() {
 	# loosen dependencies
 	sed -i '/>/s/>.*/",/g' setup.py || die
 
+	# fix Brotli dependency
+	sed -i 's/Brotli/brotlipy/g' setup.py || die
```
Comment (thesamesam) [inline net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild:61]: Depend on app-arch/brotli[python,${PYTHON_USEDEP}] instead.

**Comment by thesamesam** [comment]
You may want to split up the commits to be per-package, especially given not all of them are maintainer-needed.

Have you been able to test those deps on Py 3.7, 3.8 btw? I had marked them locally but not got around to testing it on both yet.

File: net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild (hunk 1 above)
Comment (kwhat) [reply net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild:61]: Thanks, yes that did the trick.  I was racking my head on that one for a minute.

**Comment by kwhat** [comment]
> You may want to split up the commits to be per-package, especially given not all of them are maintainer-needed.

Agreed, I as not aware it was going to touch so much stuff until it got pushed.

> Have you been able to test those deps on Py 3.7, 3.8 btw? I had marked them locally but not got around to testing it on both yet.

I have only tested minimally on Py 3.8.  I will be testing more extensively (3.7,3.8) via mitmproxy as I keep plugging away at this.  If there is other methods for testing that would make things easier, please let me know.

File: net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild (hunk 1 above)
Comment (mgorny) [reply net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild:61]: If it works with brotlipy, it's better to use that. The hyper-h2 suite requires it, and the two block each other, so a lot of people won't be able to install this if it needs `app-arch/brotli[python]`.

Comment (kwhat) [reply net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild]: It does work, just tested with Python 3.8 and it seems to function just fine. Repoman still has a lot of complaints.

**Comment by kwhat** [comment]
Also adding **dev-python/parver: Add python3_8** to fix repoman bad dependency issues.

File: dev-python/wsproto/wsproto-0.15.0-r1.ebuild [hunk 2]
```diff
@@ -18,7 +18,7 @@ IUSE="test"
 RESTRICT="!test? ( test )"
 
 RDEPEND="
```
Comment (mgorny) [inline dev-python/wsproto/wsproto-0.15.0-r1.ebuild:20]: How did you manage to get it to pass tests without listing or packaging correct dependencies?

Comment (kwhat) [reply dev-python/wsproto/wsproto-0.15.0-r1.ebuild:20]: I guess the better question is how does it currently work?

Comment (kwhat) [reply dev-python/wsproto/wsproto-0.15.0-r1.ebuild:20]: Looks like someone already took care of this upstream, maybe they fixed it.

Comment (mgorny) [reply dev-python/wsproto/wsproto-0.15.0-r1.ebuild:20]: It doesn't. But don't worry, I'll take care of packages (now) maintained by python@.

File: dev-python/kaitaistruct/kaitaistruct-0.8.ebuild [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=6
 
-PYTHON_COMPAT=( python3_6 )
+PYTHON_COMPAT=( python3_{6,7,8} )
```
Comment (mgorny) [inline dev-python/kaitaistruct/kaitaistruct-0.8.ebuild:6]: How did you test it?

Comment (kwhat) [reply dev-python/kaitaistruct/kaitaistruct-0.8.ebuild:6]: Only via mitmproxy working.  I didn't see anything else that depends on it, if there is a more through way let me know.

**Comment by thesamesam** [comment]
Please work with python@ to fix the QA issue so we can get this in.

**Comment by thesamesam** [comment]
You need to:
1. Drop Python 3.8 for now, or add it to Parver
and
2. Request stabilisation for Parver (I've done this for you).

**Comment by thesamesam** [comment]
You need to drop Python 3.8 or add it for parver. This will resolve the QA issues.

Note: to let us stabilise this in the future, I've [requested](https://bugs.gentoo.org/722562) stabilisation for parver too.

File: net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild [hunk 4]
```diff
@@ -21,18 +21,21 @@ RDEPEND="
...
 	>=dev-python/passlib-1.6.5[${PYTHON_USEDEP}]
 	>=dev-python/protobuf-python-3.6.0[${PYTHON_USEDEP}]
+	>=dev-python/publicsuffix-2.20190205[${PYTHON_USEDEP}]
 	>=dev-python/pyasn1-0.3.1[${PYTHON_USEDEP}]
 	>=dev-python/pyopenssl-17.5[${PYTHON_USEDEP}]
 	>=dev-python/pyparsing-2.1.3[${PYTHON_USEDEP}]
 	>=dev-python/pyperclip-1.6.0[${PYTHON_USEDEP}]
 	>=dev-python/requests-2.9.1[${PYTHON_USEDEP}]
 	>=dev-python/ruamel-yaml-0.15[${PYTHON_USEDEP}]
+	>=dev-python/ruamel-yaml-clib-0.1.2[${PYTHON_USEDEP}]
```
Comment (Alessandro-Barbieri) [inline net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild:38]: why this dependency?

File: dev-python/kaitaistruct/kaitaistruct-0.8.ebuild (hunk 3 above)
Comment (Alessandro-Barbieri) [reply dev-python/kaitaistruct/kaitaistruct-0.8.ebuild:6]: upstream is at 3.7

Comment (kwhat) [reply dev-python/kaitaistruct/kaitaistruct-0.8.ebuild:6]: Dropping 3.8

File: net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild (hunk 4 above)
Comment (kwhat) [reply net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild]: Removed as it is included with ruamel-yaml

**Comment by kwhat** [comment]
> Add dev-python/zstandard to test deps.
Done.

> Add dev-python/hypothesis for test dep.
Done.

> Are you sure these are not needed for runtime?
No.  How can we figure that out?

> You could use distutils_enable_tests pytest since it looks simple enough.
Enabled it, but tests did not run with USE=test

>  * DISTUTILS_USE_SETUPTOOLS value is probably incorrect
>  *   value:    DISTUTILS_USE_SETUPTOOLS=bdepend (default?)
>  *   expected: DISTUTILS_USE_SETUPTOOLS=rdepend

Added DISTUTILS_USE_SETUPTOOLS=rdepend.

**Comment by juippis** [comment]
> 
> > Are you sure these are not needed for runtime?
> 
> No. How can we figure that out?
> 

https://github.com/mitmproxy/mitmproxy/blob/a233fdd3027310239b36b0122fe778e7d867748b/setup.py#L64

Additional test deps listed here: 
https://github.com/mitmproxy/mitmproxy/blob/a233fdd3027310239b36b0122fe778e7d867748b/setup.py#L93

450414ec5c12be4b58c6487d4e679381c7905c99

**Pull request closed without merging**